viewing_party/party.py

```diff
@@ -1,23 +1,127 @@
+
+# --------- UTILITY FUNCTIONS -------------
+
+def make_list_unique(list_of_dicts):
+    #returns a list that has no duplicate dicts
+    unique_list = []
+    for dct in list_of_dicts:
+        if dct not in unique_list:
+            unique_list.append(dct)
+    return unique_list
+
+def get_friends_watched_movies(user_data):
+    friends_watched_movies = []
+    for dct in user_data["friends"]:
+        for movie in dct["watched"]:
+            friends_watched_movies.append(movie)
+    return friends_watched_movies
+
+
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (title is None) or (genre is None) or (rating is None):
+        return None
+    return {"title":title, "genre":genre, "rating":rating,}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    '''
+    Cycles through watchlist, finds matching entry, then pops that
+    entry off of watchlist and appends it to watched.
+    '''
+    for i, movie in enumerate(user_data["watchlist"]):
+        if movie["title"] == movie_title:
+            user_data["watched"].append(user_data["watchlist"].pop(i))
+    return user_data
+

```

Comment:
In the future, try to avoid modifying a list as you're looping over it. You still pass the tests here, but it can lead to some strange bugs! It would be best to modify the list after the loop is finished.